streams/src/main/java/org/apache/kafka/streams/StreamsBuilder.java

```diff
@@ -484,22 +484,23 @@ public synchronized StreamsBuilder addStateStore(final StoreBuilder builder) {
     /**
      * @deprecated use {@link #addGlobalStore(StoreBuilder, String, Consumed, ProcessorSupplier)} instead
      */
-    @SuppressWarnings("unchecked")
     @Deprecated
-    public synchronized StreamsBuilder addGlobalStore(final StoreBuilder storeBuilder,
-                                                      final String topic,
-                                                      final String sourceName,
-                                                      final Consumed consumed,
-                                                      final String processorName,
-                                                      final ProcessorSupplier stateUpdateSupplier) {
+    public synchronized <K, V>  StreamsBuilder addGlobalStore(final StoreBuilder<KeyValueStore<K, V>> storeBuilder,
```

Comment:
Even if this this public API, it's not a "change" but a bug fix, because when we call `internalStreamsBuilder.addGlobalStore()` we cast to `StoreBuilder<KeyValue>` store in the old code already.

Tying the type for the store, `Consumed` and `ProcessorSupplier` is also a bug fix, because if the types don't match, something breaks at runtime.

At least I think so -- let me know if I miss something. Fixing a public API this way is tricky...

Reply:
the super class StateStoreNode does NOT require the KeyValueStore type so is it necessary to add KeyValueStore type? Or a wildcard type is enough?

Reply:
Do we assume the stores added as global are always kv-stores? I vaguely remember for non-global source KTable we have this restriction (KIP-300) but not clear if we have the same restriction for global stores.

Reply:
My assumption was that `InternalStreamsBuilder.addGlobalStore()` that we call below applies this limitation (cf. https://github.com/apache/kafka/pull/8273/files#r391330086).

Seems this assumption was wrong. Will remove the store type.